# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/JoinNodeTransaction.h
# Commit: ce2087ddab5c
# Full Hash: ce2087ddab5c2efcbce3b52da02b3363a37d6813
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-04 09:29:13
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 1: Make each transaction class grab their members with local variable before touching the DOM tree r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69302
# ==============================================================================

diff -r 1c8115a9a684 -r ce2087ddab5c editor/libeditor/JoinNodeTransaction.h
--- a/editor/libeditor/JoinNodeTransaction.h	Fri Apr 03 10:21:33 2020 +0000
+++ b/editor/libeditor/JoinNodeTransaction.h	Fri Apr 03 08:30:37 2020 +0000
@@ -12,6 +12,7 @@
 #include "nsID.h"    // for REFNSIID
 #include "nscore.h"  // for NS_IMETHOD
 
+class nsIContent;
 class nsINode;
 
 namespace mozilla {
@@ -26,8 +27,8 @@
  */
 class JoinNodeTransaction final : public EditTransactionBase {
  protected:
-  JoinNodeTransaction(EditorBase& aEditorBase, nsINode& aLeftNode,
-                      nsINode& aRightNode);
+  JoinNodeTransaction(EditorBase& aEditorBase, nsIContent& aLeftContent,
+                      nsIContent& aRightContent);
 
  public:
   /**
@@ -35,11 +36,12 @@
    * nodes.
    *
    * @param aEditorBase     The provider of core editing operations.
-   * @param aLeftNode       The first of two nodes to join.
-   * @param aRightNode      The second of two nodes to join.
+   * @param aLeftContent    The first of two nodes to join.
+   * @param aRightContent   The second of two nodes to join.
    */
   static already_AddRefed<JoinNodeTransaction> MaybeCreate(
-      EditorBase& aEditorBase, nsINode& aLeftNode, nsINode& aRightNode);
+      EditorBase& aEditorBase, nsIContent& aLeftContent,
+      nsIContent& aRightContent);
 
   /**
    * CanDoIt() returns true if there are enough members and can join or
@@ -56,18 +58,18 @@
  protected:
   RefPtr<EditorBase> mEditorBase;
 
-  // The nodes to operate upon.  After the merge, mRightNode remains and
-  // mLeftNode is removed from the content tree.
-  nsCOMPtr<nsINode> mLeftNode;
-  nsCOMPtr<nsINode> mRightNode;
+  // The nodes to operate upon.  After the merge, mRightContent remains and
+  // mLeftContent is removed from the content tree.
+  nsCOMPtr<nsIContent> mLeftContent;
+  nsCOMPtr<nsIContent> mRightContent;
 
   // The offset into mNode where the children of mElement are split (for
   // undo). mOffset is the index of the first child in the right node.  -1
   // means the left node had no children.
   uint32_t mOffset;
 
-  // The parent node containing mLeftNode and mRightNode.
-  nsCOMPtr<nsINode> mParent;
+  // The parent node containing mLeftContent and mRightContent.
+  nsCOMPtr<nsINode> mParentNode;
 };
 
 }  // namespace mozilla